[cxx-interop] Partially specialized globals should not get imported

### Description

See the small repro case below. C++ partially specialized global variables get imported as the wrong type, and trigger a linkage on a symbol that doesn't exist.

### Reproduction

Repro case when using C++ interop, and a bridging header:
```
// bridging_header.h
template <int N, int M>
inline const int template_gcd = template_gcd<M, N % M>;

template <int N>
inline const int template_gcd<N, 0> = N;
```
and
```
// main.swift
print(template_gcd)
```

Then, this compiles fine (but shouldn't):
```
$ swiftc main.swift -import-bridging-header bridging_header.h -cxx-interoperability-mode=default -c -o main.o
# ok
```
...because the result fails to link:
```
$ xcrun swiftc main.swift -import-bridging-header bridging_header.h -cxx-interoperability-mode=default
error: link command failed with exit code 1 (use -v to see invocation)
Undefined symbols for architecture arm64:
  "__Z12template_gcdIXT_ELi0EE", referenced from:
      _main in main-1.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```


### Expected behavior

Shouldn't import; the compilation should emit a diagnostic error saying that "template_gcd" is not available because it's a partial specialization.

### Environment

$ swiftc -v                                                                                                  
Apple Swift version 6.2-dev (LLVM 6f817b270d9dadc, Swift d36b06747a54689)
Target: arm64-apple-macosx15.0

### Additional information

_No response_

--- a/lib/ClangImporter/ImportDecl.cpp
+++ b/lib/ClangImporter/ImportDecl.cpp
@@ -4638,6 +4638,11 @@ namespace {
       return result;
     }
 
+    Decl *VisitVarTemplatePartialSpecializationDecl(
+        const clang::VarTemplatePartialSpecializationDecl *decl) {
+      return nullptr;
+    }
+
     Decl *VisitImplicitParamDecl(const clang::ImplicitParamDecl *decl) {
       // Parameters are never directly imported.
       return nullptr;

--- a/test/ClangImporter/const_values_cxx.swift
+++ b/test/ClangImporter/const_values_cxx.swift
@@ -55,9 +55,6 @@ func foo() {
 
   print(MyClass().class_const_int)
   print(MyClass.class_static_const_int)
-
-  // TODO: This seems to be incorrectly imported, this test here is just to check that the compiler doesn't crash.
-  print(template_gcd)
 }
 
 // Only imported as external declarations:

--- a/test/Interop/Cxx/templates/Inputs/module.modulemap
+++ b/test/Interop/Cxx/templates/Inputs/module.modulemap
@@ -177,3 +177,8 @@ module UninstantiatableSpecialMembers {
   header "uninstantiatable-special-members.h"
   requires cplusplus
 }
+
+module VariableTemplate {
+  header "variable-template.h"
+  requires cplusplus
+}

--- a/test/Interop/Cxx/templates/Inputs/variable-template.h
+++ b/test/Interop/Cxx/templates/Inputs/variable-template.h
@@ -1,0 +1,5 @@
+template <int N, int M>
+inline const int template_gcd = template_gcd<M, N % M>;
+
+template <int N>
+inline const int template_gcd<N, 0> = N;

--- a/test/Interop/Cxx/templates/variable-template-typechecker.swift
+++ b/test/Interop/Cxx/templates/variable-template-typechecker.swift
@@ -1,0 +1,5 @@
+// RUN: %target-typecheck-verify-swift -cxx-interoperability-mode=default -I %S/Inputs
+
+import VariableTemplate
+
+let _: CInt = template_gcd // expected-error {{cannot find 'template_gcd' in scope}}

